managed/services/management/external.go

```diff
@@ -68,6 +71,18 @@ func (e *ExternalService) AddExternal(ctx context.Context, req *managementpb.Add
 		if req.AddNode != nil && runsOnNodeId == "" {
 			runsOnNodeId = nodeID
 		}
+		if req.CredentialsSource != "" {
```

Comment:
let's add additional checks for the field we removed validation

Reply:
it should be implemented for 2 cases:
1. credentials source is not passed
2. credentials source is passed, but fields aren't presented in the source.

Reply:
1. If credentials source is not passed, pmm-admin sets & sends default username.
2. Added condition: 
check if source-credentials is passed and if no username were found throws the same message as in case before removing validation (it allow us to keep backward compatibility without test/api test changes).

Reply:
API can be used directly, so we shouldn't rely on the `pmm-admin` logic.

Reply:
Got it.
I changed in every place where the field "required" was removed from API:
- proxysql
- postgresql
- mysql